frontends/mit-open/src/pages/SearchPage/SearchPage.test.tsx

```diff
@@ -200,19 +200,25 @@ describe("SearchPage", () => {
     await within(facetsContainer).findByText("Resource Type")
   })
 
-  test("Submitting text updates URL", async () => {
-    setMockApiResponses({})
-    const { location } = renderWithProviders(<SearchPage />, { url: "?q=meow" })
-    const queryInput = await screen.findByRole<HTMLInputElement>("textbox", {
-      name: "Search for",
-    })
-    expect(queryInput.value).toBe("meow")
-    await user.clear(queryInput)
-    await user.paste("woof")
-    expect(location.current.search).toBe("?q=meow")
-    await user.click(screen.getByRole("button", { name: "Search" }))
-    expect(location.current.search).toBe("?q=woof")
-  })
+  test.each([{ withPage: false }, { withPage: true }])(
+    "Submitting text updates URL",
+    async ({ withPage }) => {
+      setMockApiResponses({})
+      const urlQueryString = withPage ? "?q=meow&page=2" : "?q=meow"
+      const { location } = renderWithProviders(<SearchPage />, {
+        url: urlQueryString,
+      })
```

Comment:
Could also just put the initial URL in the parameterization. Then adding more cases is easy, and removes some conditionals from the test. (*This is a very minor suggestion. Feel free to ignore.*)

```suggestion
  test.each([{ initialUrl: "?q=meow&page=2" }, { initialUrl: "?q=meow" }])(
    "Submitting text updates URL",
    async ({ initialUrl }) => {
      setMockApiResponses({})
      const { location } = renderWithProviders(<SearchPage />, {
        url: initialUrl,
      })
```